### PR TITLE
GEOMESA-2274 Supporting query reprojections

### PR DESCRIPTION
**Pull request by elahrvivaz**
Signed-off-by: Emilio Lahr-Vivaz <elahrvivaz@ccri.com>

### CODE REVIEW

File: geomesa-utils/src/main/scala/org/locationtech/geomesa/utils/geometry/Geometry.scala [hunk 1]
```diff
@@ -11,9 +11,13 @@ package org.locationtech.geomesa.utils.geometry
 import com.vividsolutions.jts.geom.Polygon
 import org.geotools.geometry.jts.JTS
 
+@deprecated
 object Geometry {
+
+  @deprecated
   val noPolygon : Polygon  = null
 
+  @deprecated
```
Comment (jnh5y) [inline geomesa-utils/src/main/scala/org/locationtech/geomesa/utils/geometry/Geometry.scala]: Can you add some details about the deprecation?  Is there something preferred in its place?

**Comment by jnh5y** [comment]
Do we have any ContentDataStores left?  If so, how does this interact with 'canReproject'?

File: geomesa-utils/src/main/scala/org/locationtech/geomesa/utils/geometry/Geometry.scala (hunk 1 above)
Comment (elahrvivaz) [reply geomesa-utils/src/main/scala/org/locationtech/geomesa/utils/geometry/Geometry.scala]: no, i just noticed that we don't actually use it anywhere. do you think it's worth keeping around?

**Comment by elahrvivaz** [comment]
we still use content data store in arrow-datastore, fsds-datastore, cq-engine-datastore, and stream-datastore, but none of those use the query planner.

**Comment by jnh5y** [comment]
Ok, so since we aren't using the abstract ContentDataStore, then it is up to use to handle the reprojections?  (And hence this fixes that?)

**Comment by elahrvivaz** [comment]
right

**Pull request merged**